src/python-bindings/htcondor2/loose_functions.cpp

```diff
@@ -257,3 +257,57 @@ _set_ready_state( PyObject *, PyObject * args ) {
 
 	Py_RETURN_NONE;
 }
+
+
+static PyObject *
+_send_generic_payload_command( PyObject *, PyObject * args ) {
+	const char * ad_str = nullptr;
+	const char * address = nullptr;
+	long command = -1;
+
+	if(! PyArg_ParseTuple( args, "zlz", & address, & command, & ad_str )) {
+		// PyArg_ParseTuple() has already set an exception for us.
+		return nullptr;
+	}
```

Comment:
You're serializing a ClassAd to a string and then parsing it back into an object. I see with the _send_command() function that you should be able pass the ClassAd object from Python into the C++ code, which is more efficient.